From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rong Bao <webmaster@csmantle.top>
Date: Fri, 2 Oct 2026 03:47:44 +0000
Subject: [PATCH 02/11] Bug 2077374 - Add missing MOZ_RUNINIT attributes to
 global vars in mozglue/build/{mips,ppc}.cpp. r=glandium!

Differential Revision: https://phabricator.services.mozilla.com/D330303
---
 mozglue/build/mips.cpp | 4 +++-
 mozglue/build/ppc.cpp  | 8 +++++---
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/mozglue/build/mips.cpp b/mozglue/build/mips.cpp
index 5772441e1255a8b48628269f477aac0cbcd143b3..f2d6f9806ee13efe5eb1a004bb37abf4bfbe7e7f 100644
--- a/mozglue/build/mips.cpp
+++ b/mozglue/build/mips.cpp
@@ -6,6 +6,8 @@
 
 #include "mips.h"
 
+#include "mozilla/Attributes.h"
+
 #include <stdio.h>
 #include <string.h>
 
@@ -36,6 +38,6 @@ static bool check_loongson3(void) {
 
 namespace mozilla {
 namespace mips_private {
-bool isLoongson3 = check_loongson3();
+MOZ_RUNINIT bool isLoongson3 = check_loongson3();
 }  // namespace mips_private
 }  // namespace mozilla
diff --git a/mozglue/build/ppc.cpp b/mozglue/build/ppc.cpp
index cc7626f7609f3c790156ce2789ec86a10be3af1a..1e59d23802965e0d0dd473ab6cb67150b487704f 100644
--- a/mozglue/build/ppc.cpp
+++ b/mozglue/build/ppc.cpp
@@ -7,6 +7,8 @@
 
 #include "ppc.h"
 
+#include "mozilla/Attributes.h"
+
 #if defined(XP_LINUX)
 // Use the getauxval() function if available.
 // ARCH_3_00 wasn't defined until glibc 2.23, so include just in case.
@@ -53,8 +55,8 @@ static signed get_ppc_cpu_flags(void) {
 
 namespace mozilla {
 namespace ppc_private {
-bool vmx_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VMX);
-bool vsx_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VSX);
-bool vsx3_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VSX3);
+MOZ_RUNINIT bool vmx_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VMX);
+MOZ_RUNINIT bool vsx_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VSX);
+MOZ_RUNINIT bool vsx3_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VSX3);
 }  // namespace ppc_private
 }  // namespace mozilla
